func_tests/test_client.py

```diff
@@ -0,0 +1,21 @@
+from PyQt5.QtCore import Qt
+
+
+from securedrop_client.gui.main import Window
+from securedrop_client.gui.widgets import LoginDialog
+
+
+def test_login(qtbot, mocker):
+    """
+    We see an error if incomplete credentials are supplied to the login dialog.
+    """
+    w = Window()
+    login_dialog = LoginDialog(w)
+    login_dialog.error_bar.set_message = mocker.MagicMock()
+    login_dialog.show()
+    qtbot.addWidget(login_dialog)
+    assert login_dialog.error_bar.error_status_bar.text() == ""
+    qtbot.keyClicks(login_dialog.username_field, "journalist")
+    qtbot.mouseClick(login_dialog.submit, Qt.LeftButton)
+    assert login_dialog.error_bar.set_message.call_count == 1
+    assert login_dialog.error_bar.isVisible()
```

Comment:
I think we get the most value from these functional tests if we mock as minimally as possible, e.g. if we can assert the test string that will appear on the error bar instead of mocking `set_message`, I think that's preferable since we have tests that mock out these individual functions in our unit tests, so if we have functional tests that perform minimal mocking (and are thus minimally tied to the underlying implementation) then we get a big win.

Reply:
Now `vcrpy` is working, we don't mock anything. :-)